service/src/main/java/bio/terra/tanagra/service/accesscontrol/impl/AouAccessControl.java

```diff
@@ -0,0 +1,160 @@
+import bio.terra.tanagra.service.accesscontrol.Permissions;
+import bio.terra.tanagra.service.accesscontrol.ResourceCollection;
+import bio.terra.tanagra.service.accesscontrol.ResourceId;
+import bio.terra.tanagra.service.accesscontrol.ResourceType;
+import bio.terra.tanagra.service.auth.UserId;
+import com.google.common.collect.ImmutableList;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import javax.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AouAccessControl implements AccessControl {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(AouAccessControl.class);
+
+  private String basePath;
+
+  @Override
+  public void initialize(List<String> params, String basePath, String oauthClientId) {
+    if (basePath == null || oauthClientId == null) {
+      throw new IllegalArgumentException(
+          "Base URL and OAuth client id are required for VUMC admin service API calls");
+    }
+    this.basePath = basePath;
+  }
+
+  @Override
+  public String getDescription() {
+    return "Check WorkspaceAccessLevel of the user for the workspace in All of Us Research Workbench";
+  }
+
+  @Override
+  public boolean isAuthorized(UserId user, Permissions permissions, @Nullable ResourceId resource) {
+    // AoU has access levels for studyId which is same as AoU workspaceNamespace
+    if (permissions.getType().equals(ResourceType.STUDY)
+        && permissions.getActions().contains(Action.CREATE)) {
+      // AoU is calling Tanagra to create study - do not check authorization
+      return true;
+    } else {
+      try {
+        String accessLevel = getWorkspaceAccess(user, resource);
+        LOGGER.info(
+            "User access level for workspace [{}] is [{}]", resource.getStudy(), accessLevel);
+        switch (accessLevel) {
+          case "OWNER":
+            return true;
+          case "WRITER":
+            if (permissions.getType().equals(ResourceType.STUDY)
+                && (permissions.getActions().contains(Action.UPDATE)
+                    || permissions.getActions().contains(Action.DELETE))) {
+              return false;
+            }
+            return true;
+          case "READER":
+            return permissions.getActions().contains(Action.QUERY_COUNTS)
+                || permissions.getActions().contains(Action.QUERY_INSTANCES)
+                || permissions.getActions().contains(Action.READ);
+          default:
+            return false;
+        }
+      } catch (Exception ex) {
+        LOGGER.error(
+            "Workspace [{}] does not exist or user [{}] does not have permission to use it.",
+            resource.getStudy(),
+            user.getEmail(),
+            ex);
+        return false;
+      }
+    }
+  }
+
+  @Override
+  public ResourceCollection listAllPermissions(
```

Comment:
This method needs to return a list of the resources that the user can see, along with the permissions they have for each. e.g. For studies, a list of workspace/study id - workspace role/tanagra permissions.